packages/beacon-node/src/chain/stateCache/persistentCheckpointsCache.ts

```diff
@@ -575,6 +589,84 @@ export class PersistentCheckpointStateCache implements CheckpointStateCache {
     return Array.from(this.cache.keys());
   }
 
+  /**
+   * Prune or persist checkpoint states in an epoch, see the description in `processState()` function
+   */
+  private async processPastEpoch(
```

Comment:
nothing new for this function, it's moved from `processState()` to make that function shorter